tests/unit_tests/test_wandb_run.py

```diff
@@ -454,3 +455,29 @@ def test_settings_unexpected_args_telemetry(runner, relay_server, capsys, user):
             # whose field 2 corresponds to unexpected arguments in Settings
             assert 2 in telemetry.get("11", [])
             run.finish()
+
+
+def test_artifact_wait_success(user, test_settings):
+    # Test artifact wait() timeout parameter
+    timeout = 2
+    leeway = 0.50
+    run = wandb.init(settings=test_settings())
+    artifact = wandb.Artifact("art", type="dataset")
+    start_timestamp = time.time()
+    run.log_artifact(artifact).wait(wait_timeout_secs=timeout)
+    elapsed_time = time.time() - start_timestamp
+    assert elapsed_time < timeout * (1 + leeway)
+    run.finish()
+
+
+@pytest.mark.parametrize("timeout", [0, 2])
+def test_artifact_wait_failure(user, test_settings, timeout):
+    # Test to expect TimeoutError when wait timeout is reached and large image
+    # wasn't uploaded yet
+    large_image = wandb.Image(np.random.randint(0, 255, (10000, 10000, 3)))
```

Comment:
Oh wow this is huge. Someone else will need to jump in here, will this test be okay to run on circle? @wandb/sdk-team

Reply:
Maybe testing with 0 is enough, even though its kind of a corner case

Reply:
yeah I had trouble getting it to timeout so I ended up using that image size. Let me try making it smaller

Reply:
Aha, a timeout of 0 means "no timeout" -- see [this condition](https://github.com/wandb/client/blob/4c87b3676c464414d31550e52564dd0c7fec577a/wandb/sdk/interface/message_future_poll.py#L46):
```python
            if timeout and start_time - now_time > timeout:
                done = True
                continue
```
`timeout=0` and `timeout=None` will both fail that check, because they're both falsey.

Maybe try a timeout of `1e-12`? Then I bet you won't need the image to be so huge.

Reply:
I believe the wait logic passed in from artifact.wait() is actually in https://github.com/wandb/wandb/blob/4c87b3676c464414d31550e52564dd0c7fec577a/wandb/sdk/interface/router.py#L35
So timeout=None still works

Reply:
Yea, I agree with @KyleGoyette -- this might sporadically blow up the Docker container that Circle assigns to the job. How about mocking some artifact code internals with some sleep statements?